### ISSUE TITLE
Map of multigraded modules removes degrees

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#2368]
Here is an example:
```m2
A = kk[x_0..x_1, DegreeRank => 2]
B = kk[a_0..a_1,y_0..y_1, DegreeRank => 4]
f = map(B, A, {B_0, B_1})
degrees A^{{1,2}}     -- {{-1, -2}}
degrees B^{{1,2,0,0}} -- {{-1, -2, 0, 0}}
degrees f A^{{1,2}}   -- {{ 0,  0, 0, 0}}, seems wrong
degrees sub(A^{{1,2}}, B) -- same
```

**Comment by DanGrayson** [comment]
If `f` were homogeneous, it would probably do something more sensible.  Try giving f a degree map.

**Comment by mahrud** [comment]
> Try giving f a degree map.

The usage of `DegreeMap` for multigraded rings is hard to use (and poorly documented), but regardless, I shouldn't need to. Here is another examples where M2 should just do the sensible thing:
```m2
A = kk[x_0..x_1, DegreeRank => 2]
R = A ** A
f = map(R, A)
degrees f A^{{1,2}} -- {{0, 0, 0, 0}}
```

**Comment by mikestillman** [comment]
I think that having it do something better by default is a good idea.

**Comment by mahrud** [comment]
Yes. Though both [map(Ring, Ring, Matrix)](https://github.com/Macaulay2/M2/blob/448a45ab372e6942254f52f511a9cb6f7de03518/M2/Macaulay2/m2/ringmap.m2#L34) and [kernel RingMap](https://github.com/Macaulay2/M2/blob/448a45ab372e6942254f52f511a9cb6f7de03518/M2/Macaulay2/m2/ringmap.m2#L177) are intimidatingly long for me. I think it might help if I broke them into smaller functions, perhaps using hooks to take care of special cases like maps involving Galois fields, then we can focus on the piece that guesses the correct degree map.

**Comment by mahrud** [comment]
What is a homogeneous ring map? The [documentation](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_is__Homogeneous.html) doesn't say.
